conans/model/build_info.py

```diff
@@ -132,6 +160,120 @@ def _get_cpp_info():
         return self.configs.setdefault(config, _get_cpp_info())
 
 
+class Component(object):
+
+    def __init__(self, parent, name):
+        self._parent = parent
+        self.name = name
+        self.deps = []
+        self._lib = None
+        self._exe = None
+        self.system_deps = []
+        self._includedirs = []
+        self._libdirs = []
+        self._resdirs = []
+        self._bindirs = []
+        self._builddirs = []
+        self.defines = []
+        self.cflags = []
+        self.cppflags = []
+        self.cxxflags = []
+        self.sharedlinkflags = []
+        self.exelinkflags = []
+
+    @property
+    def lib(self):
+        return self._lib
+
+    @lib.setter
+    def lib(self, name):
+        if self._exe:
+            raise ConanException("'.exe' is already set for this Component")
+        self._lib = name
+
+    @property
+    def exe(self):
+        return self._exe
+
+    @exe.setter
+    def exe(self, name):
+        if self._lib:
+            raise ConanException("'.lib' is already set for this Component")
+        self._exe = name
+
+    @property
+    def includedirs(self):
+        return DirList(self._parent.includedirs, self._includedirs)
```

Comment:
I am not sure that the returned includedirs should have the parent data. This is what the build system should deal with, if we are talking about cmake, we would have an interface target for the package that defines the general include dirs, and maybe that target is linked from the components.

So maybe we need to think a little bit about the consumption model first, to have a better understanding of the model.

Reply:
Yes, I am having some doubts about this as well